components/table/src/simple-columns/favorite-column.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Optional,
+  Output,
+  ViewEncapsulation,
+} from '@angular/core';
+import { DtTable } from '../table';
+import { DtSimpleColumnBase } from './simple-column-base';
+
+@Component({
+  selector: 'dt-favorite-column',
+  templateUrl: 'favorite-column.html',
+  styleUrls: ['./favorite-column.scss'],
+  preserveWhitespaces: false,
+  encapsulation: ViewEncapsulation.Emulated,
+  /*
+   * Deliberatley set to Default because the embedded view that gets created for the
+   * dtColumDef can't handle onPush and results in ChangeAfterChecked error.
+   */
+  changeDetection: ChangeDetectionStrategy.Default,
+  providers: [{ provide: DtSimpleColumnBase, useExisting: DtFavoriteColumn }],
+})
+export class DtFavoriteColumn<T> extends DtSimpleColumnBase<T> {
+  @Output() readonly change = new EventEmitter<T>();
+
+  favorite: boolean = false;
+
+  constructor(@Optional() table: DtTable<T>) {
+    super(table);
+  }
+
+  /**
+   * @internal Get data either returns a data access with the given name or calls the
+   * displayAccessor function to get the simpleData for display.
+   */
+  // tslint:disable-next-line: no-any
+  _getData(data: T): any {
+    return this.displayAccessor
+      ? this.displayAccessor(data, this.name)
+      : (data as any)[this.name]; // tslint:disable-line:no-any
+  }
+
+  onChange(data: T): void {
```

Comment:
* We try to avoid generic method names like these if possible. Maybe something like `toggleFavourite` or if it does not work `handleClick` is also possible in certain cases. See: https://barista.lab.dynatrace.org/components/coding-standards#methods
* Please also add a doc comment that further describes this method.
* I think this method does not need to be public. Can we make it internal?

Reply:
Name was changed to `_toggleFavorite, comment was added describing the method and also was set as internal